pkg/provisioner/cloudprovisioner.go

```diff
@@ -1182,5 +1474,160 @@ func pickAvailabilityZone(requirement *azdiskv1beta2.TopologyRequirement, region
 }
 
 func (c *CloudProvisioner) isPerfOptimizationEnabled() bool {
-	return c.config.NodeConfig.EnablePerfOptimization
+	return c.config.NodeConfig.EnablePerfOptimization 
+}
+
+func GetLastSegment(ID, separator string) (string, error) {
```

Comment:
please move the below functions to utils package since they are not related to cloudprovisioner